### PR TITLE
layer: add filesystem MediaType

### PR DESCRIPTION
**Pull request by Maddosaurus**
To enable mounting of node filesystems, a new MediaType, application/x.filesystem+uncompressed, is introduced. The naming follows RFC6838 standards.
As the fileystem does not provide a ReaderAt, the Reader function has been updated with a check to return an error if the layer does not have a ReaderAt.

This is tracked by https://issues.redhat.com/browse/ROX-22259

### CODE REVIEW

File: layer.go [hunk 1]
```diff
@@ -112,6 +112,13 @@ func (l *Layer) Init(ctx context.Context, desc *LayerDescription, r io.ReaderAt)
 			return fmt.Errorf("claircore: layer %v: unable to create fs.FS: %w", desc.Digest, err)
 		}
 		l.sys = sys
+	case `application/x.filesystem+uncompressed`:
+		if desc.URI == "" {
+			return fmt.Errorf("claircore: layer %v: unable to create fs.FS: No URI provided", desc.Digest)
```
Comment (RTann) [inline layer.go:117]:
```suggestion
			return fmt.Errorf("claircore: layer %v: unable to create fs.FS: no URI provided", desc.Digest)
```

File: layer.go [hunk 2]
```diff
@@ -112,6 +112,13 @@ func (l *Layer) Init(ctx context.Context, desc *LayerDescription, r io.ReaderAt)
 			return fmt.Errorf("claircore: layer %v: unable to create fs.FS: %w", desc.Digest, err)
 		}
 		l.sys = sys
+	case `application/x.filesystem+uncompressed`:
```
Comment (crozzy) [inline layer.go:115]: What is the reasoning behind this content type? (Maybe `application/x-filesystem`) is the uncompressed implied through omission?

File: layer.go [hunk 3]
```diff
@@ -196,6 +203,9 @@ func (l *Layer) Reader() (ReadAtCloser, error) {
 	if !l.init {
 		return nil, errors.New("claircore: unable to return Reader: uninitialized Layer")
 	}
+	if l.rd == nil {
```
Comment (crozzy) [inline layer.go:206]: Yeah, was going to mention this, maybe it's time to start deprecating it (future).

File: layer.go (hunk 2 above)
Comment (Maddosaurus) [reply layer.go:115]: Piotr and I weren't able to find a good candidate from existing types, so we came up with this one.
We can absolutely switch that to `application/x-filesystem`, no worries.

**Comment by vikin91** [comment]
As Matthias is off for few more days, I rebased and reworded the commit message to reflect the change in the custom media type. Is there anything that needs to be done here for the review?

**Review by crozzy — APPROVED** [review]
LGTM

**Comment by vikin91** [comment]
Rebasing before the merge

**Comment by vikin91** [comment]
@crozzy could you help merging this?

**Comment by RTann** [comment]
finally

**Pull request merged**